src/Umbraco.Web.BackOffice/Controllers/HelpController.cs

```diff
@@ -13,15 +20,44 @@ namespace Umbraco.Cms.Web.BackOffice.Controllers
     public class HelpController : UmbracoAuthorizedJsonController
     {
         private readonly ILogger<HelpController> _logger;
+        private HelpPageSettings _helpPageSettings;
 
+        [Obsolete("Use constructor that takes IOptions<HelpPageSettings>")]
         public HelpController(ILogger<HelpController> logger)
+            : this(logger, StaticServiceProvider.Instance.GetRequiredService<IOptionsMonitor<HelpPageSettings>>())
+        {
+        }
+
+        [ActivatorUtilitiesConstructor]
+        public HelpController(
+            ILogger<HelpController> logger,
+            IOptionsMonitor<HelpPageSettings> helpPageSettings)
         {
             _logger = logger;
+
+            ResetHelpPageSettings(helpPageSettings.CurrentValue);
+            helpPageSettings.OnChange(ResetHelpPageSettings);
+        }
+
+        private void ResetHelpPageSettings(HelpPageSettings settings)
+        {
+            _helpPageSettings = settings;
         }
 
         private static HttpClient _httpClient;
+
         public async Task<List<HelpPage>> GetContextHelpForPage(string section, string tree, string baseUrl = "https://our.umbraco.com")
         {
+            if (IsAllowedUrl(baseUrl) is false)
+            {
+                _logger.LogError($"The following URL is not listed in the allowlist for HelpPage in web.config: {baseUrl}");
```

Comment:
Shouldn't this message refer to `appsettings.json` (or more specifically/correct `HelpPageSettings`, as it can also be set from code)?

Reply:
Yeah, very true, good catch 👍 I think I was a little too quick when migrating the code from V8 and missed that one 😄

Reply:
No worries 😉 I see it's updated in https://github.com/umbraco/Umbraco-CMS/commit/c73d0bf6a6b97ecc95f3d1e85db635ee0353deb5 👍🏻